pgmpy/inference/continuous/sampling.py

```diff
@@ -0,0 +1,588 @@
+        if lsteps is None:
+            lsteps = int(max(1, round(trajectory_length / stepsize, 0)))
+
+        grad_bar, _ = self.model.get_gradient_log_pdf(position_bar, self.grad_log_pdf)
+
+        for _ in range(lsteps):
+            position_bar, momentum_bar, grad_bar =\
+                self.simulate_dynamics(self.model, position_bar, momentum_bar,
+                                       stepsize, self.grad_log_pdf, grad_bar).get_proposed_values()
+
+        acceptance_prob = self._acceptance_prob(position, position_bar, momentum, momentum_bar)
+
+        # Metropolis acceptance probability
+        alpha = min(1, acceptance_prob)
+
+        # Accept or reject the new proposed value of position, i.e position_bar
+        if np.random.rand() < alpha:
+            position = position_bar.copy()
+            self.accepted_proposals += 1.0
+
+        return position, alpha
+
+    def sample(self, initial_pos, num_samples, trajectory_length, stepsize=None, return_type='recarray'):
+        """
+        Method to return samples using Hamiltonian Monte Carlo
+
+        Parameters
+        ----------
+        initial_pos: A 1d array like object
+            Vector representing values of parameter position, the starting
+            state in markov chain.
+
+        num_samples: int
+            Number of samples to be generated
+
+        trajectory_length: int or float
+            Target trajectory length, stepsize * number of steps(L),
+            where L is the number of steps taken per HMC iteration,
+            and stepsize is step size for splitting time method.
+
+        stepsize: float , defaults to None
+            The stepsize for proposing new values of position and momentum in simulate_dynamics
+            If None, then will be choosen suitably
+
+        return_type: string, defaults to 'recarray'
+            A string to get output in different types. Allowed options are
+            'dataframe' for output samples in form of pandas.DataFrame
+            'recarray' for output samples in form of numpy.recarray array
+
+
+        Returns
+        -------
+        list: A list of numpy array type objects containing samples
+
+        Examples
+        --------
+        >>> from pgmpy.inference.continuous import HamiltonianMC as HMC, GradLogPDFGaussian, ModifiedEuler
+        >>> from pgmpy.models import JointGaussianDistribution as JGD
+        >>> import numpy as np
+        >>> mean = np.array([1, -1])
+        >>> covariance = np.array([[1, 0.2], [0.2, 1]])
+        >>> model = JGD(['x', 'y'], mean, covariance)
+        >>> sampler = HMC(model=model, grad_log_pdf=GradLogPDFGaussian, simulate_dynamics=ModifiedEuler)
+        >>> samples = sampler.sample(np.array([1, 1]), num_samples = 5,
+        ...                          trajectory_length=6, stepsize=0.25, return_type='dataframe')
+        >>> samples
+                       x              y
+        0  4.940656e-324  4.940656e-324
```

Comment:
@ankurankan The first sample shouldn't have been this value, I'm really confused as why this is happening in both with `numpy.recarray` and `pandas.DataFrame`. I tried to printed values before assigning it in numpy.recarray and it was different. Can you suggest something for this ? Or otherwise should I remove the first sample and increase `num_samples` by  one to accommodate for this removal.